lib/gringotts/gateways/securion_pay.ex

```diff
@@ -0,0 +1,217 @@
+          }
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  ## Example 2
+      iex> amount = Money.new(20, :USD}
+      iex> opts = [config: [:secret_key: "c2tfdGVzdF9GZjJKcHE1OXNTV1Q3cW1JOWF0aWk1elI6"], customerId: "cust_zpYEBK396q3rvIBZYc3PIDwT"]
+      iex> card = "card_LqTT5tC10BQzDbwWJhFWXDoP"
+      iex> result = Gringotts.Gateways.SecurionPay.authorize(amount, card, opts)
+
+  """
+  @spec authorize(Money.t(), CreditCard.t() | {}, keyword) :: {:ok | :error, Response}
+
+  def authorize(amount, %CreditCard{} = card, opts) do
+    header = [{"Authorization", "Basic " <> opts[:config][:secret_key]}]
+    token_id = create_token(card, header)
+
+    token_id
+    |> create_params(amount, false)
+    |> commit(:post, "charges", header)
+    |> respond
+  end
+
+  def authorize(amount, card, opts) do
+    header = [{"Authorization", "Basic " <> opts[:config][:secret_key]}]
+
+    params = create_params({card, opts[:customerId]}, amount, false)
+
+    params
+    |> commit(:post, "charges", header)
+    |> respond
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+
+  # Creates the parameters for authorise function when 
+  # cardId and customerId is provided.
+  @spec create_params({}, Money.t(), boolean) :: {[]}
+  defp create_params({cardId, customerId}, amount, captured) do
+    {currency, value, _, _} = Money.to_integer_exp(amount)
+
+    [
+      {"amount", value},
+      {"currency", to_string(currency)},
+      {"card", cardId},
+      {"captured", "#{captured}"},
+      {"customerId", customerId}
+    ]
+  end
+
+  # Creates the parameters for authorise when token is provided.
+  @spec create_params(Integer.t(), Money.t(), boolean) :: {[]}
+  defp create_params(token, amount, captured) do
+    {currency, value, _, _} = Money.to_integer_exp(amount)
+
+    [
+      {"amount", value},
+      {"currency", to_string(currency)},
+      {"card", token},
+      {"captured", "#{captured}"}
+    ]
+  end
+
+  # Makes the request to SecurionPay's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+  defp commit(params, method, path, header) do
+    HTTPoison.request(method, "#{@base_url}#{path}", {:form, params}, header)
+  end
+
+  # Parses SecurionPay's response and returns a `Gringotts.Response` struct
+  # in a `:ok`, `:error` tuple.
+  @spec respond(term) :: {:ok | :error, Response}
+
+  defp respond({:ok, %{status_code: 200, body: body}}) do
+    parsed_body = Poison.decode!(body)
+
+    {:ok,
+     %{
+       success: true,
+       id: Map.get(parsed_body, "id"),
+       token: Map.get(parsed_body["card"], "id"),
+       status_code: 200,
+       gateway_code: 200,
```

Comment:
1. `gateway_code` cannot be set like this. It is the gateway's response code "as-is". If there's no response code, let it be `nil`.
I'm sure SecurionPay returns an error code that can go in this field.
2. Also, you don't have to explicitly set the `nil` values in a struct.